spec/support/authentication.rb

```diff
@@ -50,3 +56,39 @@ def create_user(login)
   }
   let(:api_key) { the_user.credentials.api_key }
 end
+
+shared_context "create host" do
+  def create_host(host_login)
+    id = "rspec:host:#{host_login}"
+    host_role = Role.create(role_id: id).tap do |role|
+      options = { role: role }
+      Credentials.create(options)
+      role.reload
+    end
+    Resource.create(resource_id: id, owner_id: id).tap do |resource|
+      resource.reload
+      host_role.reload
+    end
+
+    return host_role
+  end
+
+  let(:host_login) { "default-host-login" }
+  # 'let!' always runs before the example; 'let' is lazily evaluated.
+  let!(:the_host) {
+    create_host(host_login)
+  }
+  let(:host_api_key) { the_host.credentials.api_key }
+end
+
+shared_context "host authenticate Basic" do
+  let(:params) { { account: host_account, authenticator: authenticator } }
+  let(:basic_auth_header) {
+    basic = Base64.strict_encode64(['host/' + host_login, basic_password].join(':'))
+    puts(basic)
```

Comment:
Looks like a debug statement you forgot to remove?

Reply:
removed